fn/slice.go

```diff
@@ -258,3 +284,68 @@ func ForEachConc[A, B any](f func(A) B,
 
 	return bs
 }
+
+// CollectOptions collects a list of Options into a single Option of the list of
+// Some values in it. If there are any Nones present it will return None.
+func CollectOptions[A any](options []Option[A]) Option[[]A] {
```

Comment:
For my own context, what's an example use case for this?

Reply:
Let's say you want to dispatch a number of operations that all return an option, but you only want to succeed and move the computation forward if they *all* succeed. This allows us do do a functor inversion between the list functor and the option functor in a way that ensures all of the options came back as some. This is distinct from something like TrimNones where we always succeed but prune the failed computations.

CollectResults will be more useful in practice but the semantics of Result and Option usually mirror each other. The only difference between them is that Option is a monad with a *canonical* annihilator and Result is not since there is no sensible "default error". All of their other semantics are identical though.